src/core/search/sagas/searchOrganizationDataSets.js

```diff
@@ -69,44 +58,36 @@ function* searchOrganizationDataSetsWorker(action :SequenceAction) :Saga<WorkerR
       throw new Error(ERR_MISSING_ORG);
     }
 
-    const propertyTypes :Map<UUID, PropertyType> = yield select(selectPropertyTypes(REQUIRED_PROPERTY_TYPES));
-    const propertyTypeIds :Map<FQN, UUID> = propertyTypes.map((propertyType) => propertyType.type).flip();
-    if (propertyTypeIds.count() !== REQUIRED_PROPERTY_TYPES.length) {
-      throw new Error(ERR_MISSING_PROPERTY_TYPE);
-    }
-
     const constraints = [{
       constraints: [{
         searchTerm: query,
       }],
     }];
 
-    const flags = entitySetFlags.filter(isDefined);
-    if (flags.length === 0) {
+    const searchFlags = flags.filter(isDefined);
+    if (searchFlags.length === 0) {
       constraints.push({
         constraints: [
-          { searchTerm: `NOT(entity.${propertyTypeIds.get(FQNS.OL_FLAGS)}:${EntitySetFlagTypes.AUDIT})` },
+          { searchTerm: `NOT(dataset.metadata.flags:${EntitySetFlagTypes.AUDIT})` },
         ],
       });
     }
     else {
-      flags.forEach((flag :?EntitySetFlagType) => {
-        if (flag) {
-          constraints.push({
-            constraints: [
-              { searchTerm: `entity.${propertyTypeIds.get(FQNS.OL_FLAGS)}:${flag}` },
-            ],
-          });
-        }
+      searchFlags.forEach((flag) => {
+        constraints.push({
+          constraints: [
+            { searchTerm: `dataset.metadata.flags:${flag}` },
+          ],
+        });
       });
     }
 
     const response :WorkerResponse = yield call(
-      searchEntitySetDataWorker,
-      searchEntitySetData({
+      searchDataSetMetadataWorker,
+      searchDataSetMetadata({
         constraints,
-        entitySetIds: [organization.metadataEntitySetIds.datasets],
         maxHits,
+        organizationIds: [organizationId],
         start,
       }),
     );
```

Comment:
I'm getting 400s for this that are telling me that `searchTerm` is missing and `constraints` doesn't belong here. `lattice-js` validates these errors.

Reply:
You might not be on the latest version of `lattice` if you're getting validation errors, but the endpoint doesn't exist yet because my pull request has not been merged:

https://github.com/openlattice/openlattice/pull/211

You can still test locally by removing `constraints` and passing `searchTerm: query` instead.